### ISSUE TITLE
Using private attr_writer with ||= doesn't work properly

### ISSUE THREAD

**Issue by deepj** [oracle/truffleruby#1673]
To reproduce:

```ruby
class Demo
  attr_reader :priv_attr

  def start
    self.priv_attr ||= 'Something'
    self
  end

  private

  attr_writer :priv_attr
end

puts Demo.new.start.priv_attr
```

TruffleRuby 1.0.0 RC15

```
priv_attr.rb:5:in `start': private method `priv_attr=' called for Demo (NoMethodError)
Did you mean?  priv_attr
	from priv_attr.rb:14:in `<main>'
```

MRI 2.6.2

```
Something
```

**Comment by deepj** [comment]
Hm, relative to ?https://github.com/oracle/truffleruby/issues/1489?

**Comment by deepj** [comment]
@eregon hello, is it possible to take a look at this? This one caused a blocker in one of my small apps

**Comment by chrisseaton** [comment]
He's away - I'll look at it.

**Comment by deepj** [comment]
@chrisseaton Please, make this a priority

**Comment by chrisseaton** [comment]
It's an issue with the desugaring of `||=` I believe.

I simplified to this

```ruby
class Demo

  def test
    self.priv_attr ||= 14 # just = 14 would be fine
  end

  attr_reader :priv_attr

  private

  attr_writer :priv_attr

end

demo = Demo.new
demo.test
p demo.priv_attr
```

And then to:

```ruby
class Demo

  def foo
    self.bar ||= 14 # just = 14 would be fine
  end

  def bar
  end

  private

  def bar=(x)
  end

end

demo = Demo.new
demo.foo
```

**Comment by pitr-ch** [comment]
I fixed this in d407cfdabab436451e924e2785defe60722138cb, it will be in the next release.
@deepj Thank you for reporting the issue!
